data_structures/static_rtree.hpp

```diff
@@ -1170,18 +485,11 @@ class StaticRTree
         BOOST_ASSERT_MSG(leaves_stream.good(), "Reading from leaf file failed.");
```

Comment:
Clearing the goodbit is a horrible idea --- there is a reason a stream is in a bad state, it does not make sense to simply ignore it! Also, we don't throw or otherwise handle errors here, and the callsite ^ does neither.